### PR TITLE
Add warnings to users when identifier is fuzzy matched

### PR DESCRIPTION
**Pull request by Casper-Guo**
I think we should issue a warning to users when they provide an invalid input that is fuzzy corrected, just for awareness.

### CODE REVIEW

File: fastf1/plotting.py [hunk 1]
```diff
@@ -286,6 +289,12 @@ def driver_color(identifier: str) -> str:
             ratio = fuzz.ratio(identifier, existing_key)
             key_ratios.append((ratio, existing_key))
         key_ratios.sort(reverse=True)
+        if key_ratios[0][0] != 100:
+            _logger.warning(
+                ("Correcting invalid user input "
+                 f"{identifier} to {key_ratios[0][1]}."
```
Comment (theOehrly) [inline fastf1/plotting.py:295]: For readability of the output, quotes around the two variables are probably preferable.
``f"'{identifier}' to ...``

**Comment by theOehrly** [comment]
``Events.get_event_by_name`` might benefit from a similar warning in the ``else`` where fuzzy matching is done. Or did you have a reason for specifically only adding this to the plotting functions?

**Comment by Casper-Guo** [comment]
I don't know the appropriate condition for emitting a warning in `Events.get_event_by_name`.

For example, supplying "Italian" to that function will return the Italian Grand Prix correctly. Based on my reading of the implementation, the `EventName` entry is modified to `Italian `. Since `fuzz.ratio` is sensitive to whitespace, this is not a 100% match. A warning here would just be confusing.

We can maybe use `fuzz.partial_ratio` instead and emit a warning when there's no 100% match, although I think that might not be bulletproof either.

Or we can get rid of all whitespaces in the strings first before comparing them.

**Comment by theOehrly** [comment]
Are you interested in figuring that out and changing the fuzzy matching code there to make it work? 
Else I'll merge this as it is.

**Comment by Casper-Guo** [comment]
I'm pretty busy the next few days so a thorough investigation with a test suite will have to wait until next weekend at least. I suppose this is not urgent tho and I am happy to take a look at it.

**Comment by theOehrly** [comment]
Actually, with me working on the complete overhaul of the plotting code (to be done soon, hopefully), I'd be happy if I could rebase onto the master with these changes applied.

So if you are happy with that, I'd prefer to merge these changes. And then you could make a separate PR for the other changes without any time pressure.

**Comment by Casper-Guo** [comment]
Sounds good to me

**Pull request merged**